# CÂU 10: PHÂN TÍCH GIÁ TRỊ ĐƠN HÀNG THEO KẾ HOẠCH TRẢ GÓP

**1. Yêu cầu bài toán:**
Trong file `payments.csv`, tìm kế hoạch trả góp (số kỳ) có giá trị thanh toán trung bình trên mỗi đơn hàng là cao nhất.

**2. Phương pháp tiếp cận & Suy luận:**
* **Bước 1:** Tải dữ liệu các giao dịch thanh toán từ file `payments.csv`.
* **Bước 2:** Sử dụng phương pháp gom nhóm (`groupby`) dữ liệu dựa trên cột số kỳ trả góp.
* **Bước 3:** Tính giá trị trung bình (`mean`) của cột giá trị thanh toán cho từng nhóm.
* **Bước 4:** Sắp xếp kết quả theo thứ tự giảm dần để tìm ra số kỳ trả góp đứng top 1.

**3. Lý do chọn phương pháp:**
* **Sử dụng `groupby` và `mean()`:** Đây là bài toán tổng hợp thống kê cơ bản. Ta cần biết "giá trị đại diện" của một đơn hàng khi khách chọn trả 1 cục so với trả nhiều kỳ. Hàm `mean()` là phù hợp nhất để san phẳng các biến động (outliers) và đưa ra con số trung bình phản ánh đúng hành vi chi tiêu của từng nhóm.

In [1]:
# Import thư viện
import pandas as pd
from google.colab import drive

# Kết nối với Google Drive
drive.mount('/content/drive')

Mounted at /content/drive


**Thực hiện Bước 1 & 2: Tải dữ liệu và Kiểm tra cấu trúc**

Ta nạp dữ liệu thanh toán vào và kiểm tra xem tên cột lưu số kỳ trả góp và giá trị thanh toán chính xác là gì.

In [3]:
# 1. Khai báo đường dẫn
file_path = '/content/drive/MyDrive/datathon-2026-round-1/payments.csv'

# 2. Đọc dữ liệu
df_payments = pd.read_csv(file_path)

# Hiển thị 5 dòng đầu tiên
# (Thường tên cột sẽ là 'payment_installments' và 'payment_value')
df_payments.head()

,order_id,payment_method,payment_value,installments
0,1,credit_card,7967.54,3
1,2,cod,71163.75,1
2,3,credit_card,33660.99,3
3,4,credit_card,53196.25,3
4,6,paypal,1597.84,1


**Thực hiện Bước 3 & 4: Tính toán và Xếp hạng**

Gom nhóm dữ liệu theo số kỳ trả góp, tính trung bình số tiền khách đã trả, sau đó tìm ra đỉnh cao nhất.

In [4]:
# Tính giá trị trung bình theo số kỳ trả góp (cột 'installments')
avg_payment_by_installment = df_payments.groupby('installments')['payment_value'].mean().sort_values(ascending=False)

print("Giá trị thanh toán trung bình theo số kỳ trả góp:")
# Chỉ in ra 15 dòng đầu để tránh bảng quá dài
print(avg_payment_by_installment.head(15))

# Lấy số kỳ trả góp mang lại giá trị cao nhất
best_plan = avg_payment_by_installment.idxmax()
best_value = avg_payment_by_installment.max()

print(f"\n--> Kế hoạch trả góp có giá trị trung bình cao nhất là: {best_plan} kỳ (với {best_value:.2f})")

Giá trị thanh toán trung bình theo số kỳ trả góp:
installments
6     24446.654403
3     24399.635486
12    24245.772694
1     24113.274166
2       708.473729
Name: payment_value, dtype: float64

--> Kế hoạch trả góp có giá trị trung bình cao nhất là: 6 kỳ (với 24446.65)


**KẾT LUẬN CUỐI CÙNG:**
Dựa trên kết quả chạy code phân tích trung bình giá trị thanh toán theo từng nhóm trả góp, kế hoạch mang lại giá trị cao nhất là **6 kỳ** (đạt khoảng 24446.65).

Đối chiếu với các phương án của đề bài:

A) 1 kỳ (trả một lần)

B) 3 kỳ

C) 6 kỳ

D) 12 kỳ

**=> Đáp án được chọn là: C**